File: QE-selenium/python/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-attributes")
    print(driver.title)
    ActionChains(driver).click(driver.find_element(By.XPATH,"//input[starts-with(@id,'full-name')]")).send_keys("Karthik").perform()
    ActionChains(driver).click(driver.find_element(By.XPATH,"//input[starts-with(@id,'full-name')]/following::input")).send_keys("kapotugari@gmail.com").perform()
    driver.find_element(By.XPATH,"//input[@type='date']").send_keys("2026-04-24")
    driver.find_element(By.XPATH,"//button[contains(text(),'Submit')]/preceding::textarea").send_keys("Huge event")
    driver.find_element(By.XPATH,"//button[contains(text(),'Submit')]").click()
    print(driver.find_element(By.XPATH,"//h3[@id='action-confirmation']").text)
    driver.quit()

File: QE-selenium/java/Activity15.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity15 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/dynamic-attributes");
            System.out.println(driver.getTitle());

            Actions actions = new Actions(driver);

            WebElement fullNameInput = driver.findElement(By.xpath("//input[starts-with(@id,'full-name')]"));
            actions.click(fullNameInput).sendKeys("Karthik").perform();


            WebElement emailInput = driver.findElement(By.xpath("//input[starts-with(@id,'full-name')]/following::input"));
            actions.click(emailInput).sendKeys("kapotugari@gmail.com").perform();

           
            driver.findElement(By.xpath("//input[@type='date']")).sendKeys("2026-04-24");

          
            driver.findElement(By.xpath("//button[contains(text(),'Submit')]/preceding::textarea"))
                  .sendKeys("Huge event");

            driver.findElement(By.xpath("//button[contains(text(),'Submit')]")).click();

            System.out.println(driver.findElement(By.xpath("//h3[@id='action-confirmation']")).getText());

        } finally {
            driver.quit();
        }
    }
}